fsspec/spec.py

```diff
@@ -375,6 +375,16 @@ def walk(self, path, maxdepth=None, topdown=True, **kwargs):
         List all files, recursing into subdirectories; output is iterator-style,
         like ``os.walk()``. For a simple list of files, ``find()`` is available.
 
+        When topdown is True, the caller can modify the dirs list in-place,
+        and walk() will only recurse into the subdirectories whose names remain in dirs.
+        Modifying dirs when topdown is False has no effect.
+
+        When topdown is True, the caller can modify the dirnames list in-place (perhaps using del or slice assignment),
```

Comment:
The CI linting failures are all about these lines being too long. @epizut If you keep each line to 88 characters then the CI will pass.